drivers/Ticker.h

```diff
@@ -106,13 +106,13 @@ class Ticker : public TimerEvent, private NonCopyable<Ticker> {
         attach(callback(obj, method), t);
     }
 
-    /** Attach a function to be called by the Ticker, specifying the interval in micro-seconds
+    /** Attach a function to be called by the Ticker, specifying the interval in microseconds
      *
      *  @param func pointer to the function to be called
      *  @param t the time between calls in micro-seconds
      *
-     *  @note setting @a t to a value shorter that it takes to process the ticker callback
-     *  will cause the system to hang. Ticker callback will be called constantly with no time
+     *  @note setting @a t to a value shorter than it takes to process the ticker callback
+     *  causes the system to hang. Ticker callback is called constantly with no time
```

Comment:
I disagree with the tense edit here. Even if the style guide says that an API "does" something, rather than "will do" something, this is talking about the ongoing (potential) future consequences of the API call (when the callbacks start happening), not the action the call takes "now", so I think it ends up less clear.

I would prefer the present tense reserved for stuff that happens during the API call, allowing the future tense to be used for stuff that happens later.

Aside from that, the statement was always a bit black-and-white - it's quite possible that the processing time isn't constant and consistently higher than the interval, so maybe some conditional is in order.

Maybe "setting t to too small a value could significantly impair performance or cause the system to hang altogether. If a ticker callback runs beyond the next scheduled call time, when it does return it will be called again immediately, leaving no time for thread scheduling".

Reply:
@aashishc1988 @AnotherButler